### PR TITLE
feat(backend): added registration_stage field

### LINKED ISSUE #102: Add a single flag on a user to indicate to the front-end which stage of the sign-up process they are on.
## Describe the feature

Currently, there is a `requires_customer_registration_completion` flag which indicates whether upon logging in a `user` should be redirected to the create-customer-view.  However, that view includes creating a customer and then a customer-user via the API.  The former uses the `RequiresCustomerRegistrationCompletion` permission to restrict customer-creation to those uses w/ the aforementioned flag set to True.  But the latter uses the `IsAdminOrManager` permission to restrict customer-user-creation to MANAGERS of customers.  This leads to a _Catch-22_: how can the user be a manager of the customer before they have been added to said customer?

## Additional context

Initially I thought about adding a second `requires_customer_user_completion` flag and permission to wrap the customer-user-creation view.  But this could start to get unweildy.

Instead I plan to refactor the code to use a single flag:  `registration_status` w/ values like "USER|CUSTOMER|CUSTOMER_USER|ORDER|NONE" and a permission class that takes one of these values.

The tricky bit will be dealing w/ "ORDER" (b/c that is a feature of **orbis** rather than **django-astrosat-users**).

### PR DESCRIPTION
**Pull request by allynt**
Added a `registration_stage` field on `user` to indicate which stage of the registration process they are in.  (This replaces the previous `requires_customer_registration_completion` field.)

This PR closes #102

### CODE REVIEW

File: astrosat_users/forms/forms_auth.py [hunk 1]
```diff
@@ -64,10 +64,10 @@ def save(self, request, **kwargs):
 
 class RegistrationForm(AllAuthRegistrationForm):
 
-    field_order = ["email", "password1", "password2", "requires_customer_registration_completion", "accepted_terms"]
+    field_order = ["email", "password1", "password2", "registration_stage", "accepted_terms"]
 
     accepted_terms = forms.BooleanField(label="Accept Terms & Conditions")
-    requires_customer_registration_completion = forms.BooleanField(label="User is Registering as part of a Team", required=False)
+    registration_stage = forms.CharField(label="What stage of the registration process is the user at", required=False)
```
Comment (marksmall) [inline astrosat_users/forms/forms_auth.py:70]: Would this not be better as an `enum`? I assume there will only be a few possible stages.

File: astrosat_users/models/models_users.py [hunk 2]
```diff
@@ -27,6 +27,16 @@ def user_avatar_path(instance, filename):
...
 
 
+class UserRegistrationStageType(models.TextChoices):
+
+    USER = "USER", _("User")
+    CUSTOMER = "CUSTOMER", _("Customer")
+    CUSTOMER_USER = "CUSTOMER_USER", _("CustomerUser")
+    ORDER = "ORDER", _("Order")
+
+    __empty__ = _("None")
```
Comment (marksmall) [inline astrosat_users/models/models_users.py:37]: Is this a standard way to define `None`. I remember work I did to update model for `Customer` I think, it had an `enum` like this for customer types, the issue wanted a `None` option, but I didn't add one and left it to the frontend as my naive way didn't work. This is just a question for my learning.

**Review by marksmall — APPROVED** [review]
Looks good, moving to a **State Machine** I completely agree with, but suggest considering making an `enum` to represent the possible states, right now it is a free text field, so I'm not really sure what states are possible, using an `enum` would make the stages explicit. That would require a fair bit of changes to this though, so I will approve the PR and leave the decision up to you.

File: astrosat_users/forms/forms_auth.py (hunk 1 above)
Comment (allynt) [reply astrosat_users/forms/forms_auth.py:70]: The field as defined in the model already uses an enum.  Therefore, validation on this form should fail if anything other than one of the predefined choices is provided.

File: astrosat_users/models/models_users.py (hunk 2 above)
Comment (allynt) [reply astrosat_users/models/models_users.py:37]: It's not needed. By default the empty value will be `None`.  This is valid if `null=True` is set on the field definition.  All that this does is to customise the _label_ that is used for form/serializer fields.  And that's not really exposed in our apps.  So I guess there's no real point to this line.  If it bothers you, I can remove it.

Comment (marksmall) [reply astrosat_users/models/models_users.py:37]: No, it just reminded me of the change I had made and wondered if this was kinda a standard thing and I had done it a bit wrong. Happy with it, so will resolve this conversation.

**Pull request merged**